File: android_amazon_java/src/main/java/com/example/android_amazon_java/adapter/PopularAdapter.java
package com.example.android_amazon_java.adapter;

import android.content.Context;
import android.util.DisplayMetrics;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.widget.ImageView;

import androidx.recyclerview.widget.RecyclerView;

import com.example.android_amazon_java.R;
import com.example.android_amazon_java.model.Popular;

import java.util.ArrayList;

public class PopularAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private Context context;
    private ArrayList<Popular> items;

    public PopularAdapter(Context context, ArrayList<Popular> items) {
        this.context = context;
        this.items = items;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_popular, parent, false);
        setLinearHeight(view);
        return new PopularViewHolder(view);
    }

    public class PopularViewHolder extends RecyclerView.ViewHolder {
        ImageView iv_popular;

        public PopularViewHolder(View view) {
            super(view);
            iv_popular = view.findViewById(R.id.iv_popular);
        }
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        Popular item = items.get(position);

        if (holder instanceof PopularViewHolder) {
            PopularViewHolder popularViewHolder = (PopularViewHolder) holder;
            popularViewHolder.iv_popular.setImageResource(item.getImage());
        }
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    private void setLinearHeight(View itemView) {
        // Get screen width programmatically
        DisplayMetrics displayMetrics = new DisplayMetrics();
        ((WindowManager) context.getSystemService(Context.WINDOW_SERVICE)).getDefaultDisplay().getMetrics(displayMetrics);
        int widthPixels = displayMetrics.widthPixels;

        // Set layout width programmatically
        ViewGroup.LayoutParams params = itemView.getLayoutParams();
        params.width = (widthPixels / 2) - dpToPx(context, 6);
        params.height = (widthPixels / 2) - dpToPx(context, 6);

        itemView.setLayoutParams(params);
    }

    private int dpToPx(Context context, int dp) {
        return (int) (dp * context.getResources().getDisplayMetrics().density);
    }
}

File: app/src/main/java/com/example/android_amazon/adapter/PopularAdapter.kt
package com.example.android_amazon.adapter

import android.content.Context
import android.util.DisplayMetrics
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.WindowManager
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.example.android_amazon.R
import com.example.android_amazon.model.Popular
import com.example.android_amazon.model.Sneaker

class PopularAdapter(var context: Context, var items: ArrayList<Popular>): RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_popular, parent, false)
        setLinearHeight(view)
        return PopularViewHolder(view)
    }

    class PopularViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var iv_popular: ImageView
        init {
            iv_popular = view.findViewById(R.id.iv_popular)
        }
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val item = items[position]

        if(holder is PopularViewHolder) {
            var iv_popular = holder.iv_popular

            iv_popular.setImageResource(item.image)
        }
    }

    override fun getItemCount(): Int {
        return items.size
    }

    private fun setLinearHeight(itemView: View) {
        // Get screen width programmatically
        val displayMetrics = DisplayMetrics()
        (context.getSystemService(Context.WINDOW_SERVICE) as WindowManager).defaultDisplay.getMetrics(displayMetrics)
        val widthPixels = displayMetrics.widthPixels

        // Set layout width programmatically
        val params: ViewGroup.LayoutParams = itemView.layoutParams
        params.width = (widthPixels / 2)
        params.height = (widthPixels / 2)

        itemView.layoutParams = params
    }

}